### DISCUSSION TITLE
hdiutil: attach failed - no mountable file systems

### DISCUSSION THREAD

**Discussion by alexventuraio** [Homebrew/discussions#5499] [Casks]
### Output of `brew config`

```shell
╰─ brew config
HOMEBREW_VERSION: 4.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 1c98a33a376e5b35ebc801d0f7d6ba7b9eb118ac
Last commit: 9 days ago
Core tap JSON: 16 Jul 18:18 UTC
Core cask tap JSON: 16 Jul 18:18 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.3.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.3/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.45.1 => /opt/homebrew/bin/git
Curl: 8.6.0 => /usr/bin/curl
macOS: 14.5-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
╰─ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  authy

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1

Warning: You have the following deprecated, official taps tapped:
  Homebrew/homebrew-cask-versions
Untap them with `brew untap`.
```

### Description of issue

I'm trying to install [Neovide](https://neovide.dev/) using the brew cask formula and I'm getting this error:
```shell
╰─ brew install --cask neovide
==> Downloading https://formulae.brew.sh/api/cask.jws.json
##################################################################################################################################################################################################################### 100.0%
==> Downloading https://github.com/neovide/neovide/releases/download/0.13.3/Neovide-aarch64-apple-darwin.dmg
Already downloaded: /Users/alex/Library/Caches/Homebrew/downloads/abdde5d9c9edafccc0347beec1156221e54e78ff7a0c999d0b85a8b473fd0539--Neovide-aarch64-apple-darwin.dmg
==> Downloading https://formulae.brew.sh/api/formula.jws.json
##################################################################################################################################################################################################################### 100.0%
All dependencies satisfied.
==> Installing Cask neovide
hdiutil: attach failed - no mountable file systems
==> Purging files for version 0.13.3 of Cask neovide
Error: Failure while executing; `/usr/bin/env hdiutil attach -plist -nobrowse -readonly -mountrandom /private/tmp/homebrew-dmg20240716-29248-ku0izz /private/tmp/homebrew-dmg20240716-29248-ku0izz/abdde5d9c9edafccc0347beec1156221e54e78ff7a0c999d0b85a8b473fd0539--Neovide-aarch64-apple-darwin.cdr` exited with 1. Here's the output:
hdiutil: attach failed - no mountable file systems
```

### Relevant casks

neovide

**Comment by j-lakeman** [comment]
I also ran into this issue multiple times. A reboot “fixed” it temporarily.
However a proper fix would be nice.

**Comment by iPrOmar** [comment]
**Confirmed Reproducible On MacOS Tahoe 26.3 (arm64) — March 2026**
Adding to this discussion as I've just completed a detailed investigation of this issue on **_macOS Tahoe 26.3 (arm64, Apple Silicon)_** with **_Homebrew 5.0.16_**, and can confirm it remains fully broken with some additional detail on the exact failure point.

**Environment**
- macOS Tahoe 26.3 (arm64)
- Homebrew 5.0.16
- hdiutil framework: 681.60.1 / driver: 681.60.1
- Test cask: VLC 3.0.23

**Exact Failure Point**
The failure occurs specifically in the **_CDR conversion and attach pipeline_**. The sequence is:
```
✅ hdiutil attach ... vlc-3.0.23-arm64.dmg       ← direct DMG attach succeeds (no error)
✅ hdiutil convert -format UDTO ... .cdr          ← DMG→CDR conversion succeeds
❌ hdiutil attach ... vlc-3.0.23-arm64.cdr        ← CDR attach FAILS
   hdiutil: attach failed - no mountable file systems
```
`hdiutil` on Tahoe 26.3 **cannot attach a CDR file that it itself just produced**. The conversion step completes without error, but the resulting `.cdr` cannot be mounted.

When attempting to attach the original `.dmg` directly, a different but related failure occurs:
```
expected CRC32 $07905C3B
hdiutil: attach failed - no mountable file systems
```

**Reproduction**
This was reproduced three independent times:

1. With a cached DMG (SHA-256 verified by Homebrew)
2. After full cache purge via brew cleanup --prune=all with a fresh download
3. After manually deleting the cached DMG and re-downloading

All three runs produced identical failures at the identical step.

**Reboot Workaround Does Not Apply**
Others in this thread have noted that a reboot temporarily resolved the issue. **This was not the case here**. 
This Mac is shut down and rebooted every day — the issue persists consistently across fresh boots, ruling out any transient system state as the cause. This suggests the regression may be more deeply rooted on Tahoe 26.3 than previously documented...

**Reported To Apple**
This has now been formally reported to Apple via both Feedback Assistant (macOS → System Frameworks → Incorrect/Unexpected Behaviour) and the Apple Product Feedback website.

**Note**
When this was previously raised with Homebrew it was correctly identified as an Apple-side regression. This comment is intended to document the Tahoe 26.3 confirmation and the fact that it has been formally escalated to Apple.
This blocks all DMG-based cask installs and upgrades on Tahoe — including common scientific computing tools. A workaround or liaison with Apple on a fix would be greatly appreciated.

**Reply by iPrOmar** [reply]
Update: Still reproducing on macOS Tahoe 26.4.1 with Homebrew 5.1.7. The hdiutil CDR attach regression remains unfixed.
Brew config as below...

**Reply by iPrOmar** [reply]
`$ brew config
HOMEBREW_VERSION: 5.1.7-74-gc256c58
ORIGIN: https://github.com/Homebrew/brew
HEAD: c256c58d27d4451a4bcac1a5df8fbf3e09c23332
Last commit: 3 hours ago
Branch: main
Core tap HEAD: 375350a85b6c8df81574c4209029271400509691
Core tap last commit: 3 weeks ago
Core tap JSON: 23 Apr 08:38 UTC
Core cask tap JSON: 23 Apr 08:38 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /var/run/com.apple.launchd.52E2XFBBhO/org.xquartz:0
HOMEBREW_DOWNLOAD_CONCURRENCY: 16
HOMEBREW_FORBID_PACKAGES_FROM_PATHS: set
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 4.0.3 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/4.0.3/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 21.0.0 build 2100
Git: 2.50.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 26.4.1-arm64
CLT: 26.4.1.0.1775747724
Xcode: 26.4.1
Metal Toolchain: N/A
Rosetta 2: false`
